### PR TITLE
Full-stack Capstone Frontend

### PR DESCRIPTION
**Pull request by Atifzada**
### skill-habour-app

[Project Requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)

[Click here for Backend PR](https://github.com/coderacheal/skill-habour-be/pull/7)

[Click here for Kanban board](https://github.com/users/coderacheal/projects/5)

[Click here for the Backend repository](https://github.com/coderacheal/skill-habour-be/)

### Authors
@Atifzada 
@coderacheal 
@Ouail-01 

### Key Features

- login/signup
- Get a Course for reservation
- Reservations and cancelations
- Only logged in User can make reservations and delete their reservations only

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
+import Reservations from './components/Reservations';
+import MyReservations from './components/MyReservations';
+import DeleteReservations from './components/DeleteReservations';
+import HomePage from './components/HomePage';
+import CourseDetails from './components/CourseDetails';
+import ReservationForm from './components/ReservationForm';
+import Authentication from './components/Authentication';
+import AddCourse from './components/AddCourse';
 
+const App = () => (
```
Comment (Meltrust) [inline src/App.js:12]: - [ ] I can't find any links for adding or deleting courses:

image

If admin privileges are required, I also can't find in the schema a way to add admin users.

After some investigation, I found that those links are commented out:

image

These are core requirements for a group of 3.  Kindly, add/finish those features. 👍

Comment (Meltrust) [inline src/App.js:12]: - [ ] Kindly, remove the old commented code found in:

```
Dashboard.jsx
HomePage.jsx
```

That way, the repo's cleanliness will be improved. 👍

File: src/components/ReservationList.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import '../styles/MyReservations.css';
+import fetchDelete from '../features/deleteSlice';
+import SignOutButton from './SignOutButton';
+
+function ReservationList() {
```
Comment (Meltrust) [inline src/components/ReservationList.jsx:7]: - [ ]  Please, kindly refactor this functional componenent to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const ReservationList= () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Atifzada ,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Nice code organization ✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/ReservationList.jsx (hunk 2 above)
Comment (Atifzada) [reply src/components/ReservationList.jsx:7]: @Meltrust Thank you for your helpful review we will make changes as per your request

File: src/App.js [hunk 3]
```diff
@@ -1,29 +1,28 @@
...
+import Reservations from './components/Reservations';
+import MyReservations from './components/MyReservations';
+import DeleteReservations from './components/DeleteReservations';
+import HomePage from './components/HomePage';
+import CourseDetails from './components/CourseDetails';
+import Authentication from './components/Authentication';
+import AddCourse from './components/AddCourse';
+import DeleteCourse from './components/DeleteCourse';
 
+const App = () => (
```
Comment (Meltrust) [inline src/App.js:12]: - [ ] OPTIONAL: If I review the project at 1440p resolution, the courses disappear and I only get one:

1440p:

image

1080p:

image

If you don't have a 144p monitor, you can reproduce the issue by zooming out:


This is just a fair warning, so you are aware of the issue.  👍

File: src/index.css [hunk 4]
```diff
@@ -1,13 +1,38 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400&display=swap');
```
Comment (Meltrust) [inline src/index.css:1]: - [ ] In the mobile view, the description form field is not consistent with the others:

image

Kindly, add some styling so it's the same as the others.

That way, the form will have more consistency and the mobile UX will improve. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Atifzada,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

- Add and delete features have been implemented with Admin role✔️
- Commented code has been removed ✔️ 
- Functions are ES6 ✔️ 





#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js (hunk 3 above)
Comment (Atifzada) [reply src/App.js:12]: I have checked it and on my screen it's working fine you can see:
image

File: src/index.css (hunk 4 above)
Comment (Atifzada) [reply src/index.css:1]: Sure we will fix it

**Review by Meltrust — APPROVED** [review]
#### Hi @Atifzada,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause



Cheers and Happy coding!👏👏👏


------

**Pull request merged**